### PR TITLE
DUCC: add convert-singularity-image command

### PR DESCRIPTION
**Pull request by janpriessnitz**
Also separate ConvertWish into ConvertWishSingularity and ConvertWishDocker

### CODE REVIEW

File: test/src/401-ducc-convert_singularity/main [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+cvmfs_run_test() {
+  echo "*** creating empty repo ***"
+  create_empty_repo $CVMFS_TEST_REPO $CVMFS_TEST_USER || return 1
+
+  echo "*** converting image to singularity ***"
+  cvmfs_ducc convert-singularity-image --repo $TEST_REPO $DOCKER_TEST_IMAGE || return 2
+
+  return 0
+}
```
Comment (jblomer) [inline test/src/401-ducc-convert_singularity/main]: A trailing newline is missing

File: test/src/401-ducc-convert_singularity/main [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+DOCKER_TEST_IMAGE="https://registry.hub.docker.com/library/centos:centos7"
+
+
+cvmfs_run_test() {
+  echo "*** creating empty repo ***"
+  create_empty_repo $CVMFS_TEST_REPO $CVMFS_TEST_USER || return 1
+
+  echo "*** converting image to singularity ***"
+  cvmfs_ducc convert-singularity-image --repo $TEST_REPO $DOCKER_TEST_IMAGE || return 2
+
```
Comment (jblomer) [inline test/src/401-ducc-convert_singularity/main]: There should be some test code here to check that the image contents actually appeared in the repository

**Comment by DrDaveD** [comment]
It's very unusual for a command line client flag to accept `true` and `false`.  It's more common to have separate flags to turn off, for example `--no-convert-docker` or `--dont-convert-docker`.  The `cvmfs_ducc convert` help message also doesn't at all make it clear that one is supposed to pass `--convert-docker=false` to turn it off.  For that matter, it also doesn't say anywhere how to pass a wish file.

**Comment by siscia** [comment]
This is actually a very good point, from an usability point of view I agree that the use of negative flags should be prefered. @janpriessnitz I would take care of it!

**Pull request merged**